doc/fgpu.design.rst

```diff
@@ -233,17 +314,16 @@ into the postprocessing kernel (see the next section).
 
 Unzipping the FFT
 ~~~~~~~~~~~~~~~~~
-From here we'll assume all transforms are complex-to-complex unless specified
-otherwise. The Cooley-Tukey algorithm allows a transform of size :math:`c =
-mn` to be decomposed into :math:`n` transforms of size :math:`m` followed by
-:math:`m` transforms of size :math:`n`. We'll refer to :math:`n` as the
+Right --- lets get practical and show how we *actually* implement this. From here we'll assume all
+transforms are complex-to-complex unless specified otherwise. Firstly, some recap: the Cooley-Tukey
+algorithm allows a transform of size :math:`N = mn` to be decomposed into :math:`n` transforms of
+size :math:`m` followed by :math:`m` transforms of size :math:`n`. We'll refer to :math:`n` as the
 "unzipping factor". We will keep it small (typically not more than 4), as the
-implementation requires registers proportional to this factor.
+implementation requires registers proportional to this factor. We are now going to go step-by-step and
+separate the input array :math:`x` into :math:`n` parts of size :math:`m` with each part operated on using a
+Fourier transform.
 
-To recap the Cooley-Tukey algorithm: let a time-domain index :math:`i` be
-written as :math:`qn + r` and a frequency-domain index :math:`k` be
-written as :math:`pm + s`. Let :math:`z^r` denote the array :math:`z_r, z_{n+r},
-\dots, z_{(m-1)n+r}`, and denote its Fourier transform by :math:`Z^r`. Then
```

Comment:
By moving these definitions to later on, you're now using them before they're defined.

Reply:
The definition follows immediately after so is it a major issue?

Reply:
Reordered to aid the flow.